packages/adminrouter/extra/src/test-harness/tests/test_common.py

```diff
@@ -0,0 +1,25 @@
+import grp
+
+import psutil
+import pytest
+
+
+# A hack to be able to parameterize a test with multiple fixtures.
+# https://github.com/pytest-dev/pytest/issues/349#issuecomment-189370273
+@pytest.fixture(
+    params=[
+        'agent_ar_process',
+        'master_ar_process'
+    ])
+def ar_process(request):
+    return request.getfuncargvalue(request.param)
+
+
+class TestNginxWorkersGroup:
+    def test_if_nginx_workers_on_master_have_correct_group(self, ar_process):
+        gid = grp.getgrnam('dcos_adminrouter').gr_gid
+        for proc in psutil.process_iter():
+            if proc.name() == 'nginx':
```

Comment:
Instead of iterating over the ps tree, we can extract the PID of the subprocess that was launched by ManagedSubprocess class (<link>). We just need to add an attribute to it, something like:
```
@property
def pid(self):
    return self._process.pid
```

And it will give as PPID of the process group. This will make your tests much more robust and explicit.